### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by Yonashenok**
## Summary 
[@Yonashenok](https://github.com/Yonashenok) I have completed the implementation of [Budget app](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) on the Budget app according to project [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md). 

The following tasks have been implemented: 
- [x] simple page with the name of your app
- [x] Login page and registration page using devise
- [x] Home page (categories page)
   -  user logs in, they are presented with the categories page
   -  user can see their name, icon and the total amount of all the transactions
   - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category
   - button "add a new category" at the bottom that brings the user to the page to create a new category
- [x] Transactions page
    - The list of transactions is presented, ordered by the most recent  
    - The top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
 - [x] "Add a new category" page
    -  The user fills out a form to create a new category, indicating their name and icon
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success
    - When the user clicks on the "Back" button (<), the user navigates to the home page
 - [x]  Add a new transaction" page
    -  he user fills out a form to create a new transaction
    - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category
- [x] Testing requirements
   -  Create unit and integration tests for all the most important components of your RoR application 

[Technical requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#technical-requirements)
- Should use devise for authentication.
- Should validate all user input to make sure that anyone with bad intentions - - cannot compromise your app.
- Can use a view template engine of your choice (.erb, .slim, .haml).
- Project should be deployed and accessible online.
- Database schema should reflect the following structure ERD:

 

In addition, we have fulfilled the following default requirements:

* Set up the repository on GitHub and use Gitflow.
* Set up Devise for authentication.
* Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
* Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
* Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
* Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
* Set up RSpec and Capybara testing libraries.

we will be  looking forward to your feedback.
Thank you!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,182 @@
...
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+>
+
+- [Live Demo](https://expensepal.onrender.com)
+
+- [Zoom](https://drive.google.com/file/d/1XKKi8Cxlu_EIRKyRbC0ClvRl0rd2OP37/view?usp=sharing)
```
Comment (mckent05) [inline README.md:59]: - Good job done with your video, but according to the project requirements, your video should `contain a highlight about a piece of the code the student finds interesting or is proud of making`. I suggest you remake the video and highlight a piece of code you were proud of implementing. Please fix this 👍 👍

File: README.md [hunk 2]
```diff
@@ -1 +1,182 @@
...
+If you like this project please give it a Start ⭐️.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- Project from [Microverse](https://www.microverse.org/) Ruby on rails Module.
+- Thanks to the Microverse team for the great curriculum.
+- Thanks to the Code Reviewer(s) for the insightful feedbacks.
+- A great thanks to My coding partner(s), morning session team, and standup team for their contributions.
```
Comment (mckent05) [inline README.md:162]: - You have done well so far, but according to the project requirements: The README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). Please acknowledge the designer whose design you used

File: app/views/catagories/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      <p class="card-text"><small class="text-muted"><%= time_ago_in_words(payment.created_at) %> ago</small></p>
+    </div>
+    </div>
+    <span class="badge bg-light text-dark rounded"><h4>$<%= payment.amount %>.00</h4></span>
+</li>
+<% end %>
+<div>
+<div class="d-flex flex-column justify-content-center  align-items-center">
+<%= link_to "Add New Payment", new_catagory_payment_path(@catagory.id), class: "btn px-5 text-white fw-bold  btnColor mt-5" %>
+  </div>
```
Comment (mckent05) [inline app/views/catagories/show.html.erb:43]: - You have done well so far, but according to the project requirements you are required to follow the design guidelines in terms of `Layout: composition and space between elements`. The `Add New Payment` button is not styled as it is in the design template. Please fix this 👍 👍 

Design template:
Cwrgawrfgawr

Your design:
- Please place your button at the button of the page as you did in the category page 😄 
rttrt

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Yonashenok, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- You have done well working on your capstone project 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/catagories/show.html.erb (hunk 3 above)
Comment (Yonashenok) [reply app/views/catagories/show.html.erb:43]: it mobile app you looking on desktop view

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
+require 'factory_bot_rails'
```
Comment (julie-ify) [inline spec/rails_helper.rb:3]: - [ ] Your tests do not run. Kindly fix them. You are requiring factory bot but this library does not exist in your Gemfile

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @Yonashenok,

Good job so far!

## Highlights!
- Good code organization and app design ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/rails_helper.rb (hunk 4 above)
Comment (Yonashenok) [reply spec/rails_helper.rb:3]: image

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @Yonashenok ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Tests are running and passing as expected ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**